sphinx.deprecation.RemovedInSphinx50Warning: The 'ignore' argument to prepare_docstring() is deprecated

I went to update the Sphinx requirement in `requirements-dev.txt`, but it causes tests to fail:

```
______________________________________________________ test_show_docstring ______________________________________________________

tmpdir = local('/tmp/pytest-of-willkg/pytest-3/test_show_docstring0')

    def test_show_docstring(tmpdir):
        rst = dedent(
            """\
        .. autocomponent:: test_sphinxext.ComponentWithDocstring
           :show-docstring:
    
        """
        )
    
>       assert parse(tmpdir, rst) == dedent(
            """\
            component test_sphinxext.ComponentWithDocstring
    
               This component is the best.
    
               The best!
    
               Options:
                  **user** (*str*) --
            """
        )

tests/test_sphinxext.py:248: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/test_sphinxext.py:40: in parse
    fakesphinx.builder.build_all()
../../venvs/everett/lib/python3.7/site-packages/sphinx/builders/__init__.py:260: in build_all
    self.build(None, summary=__('all source files'), method='all')
../../venvs/everett/lib/python3.7/site-packages/sphinx/builders/__init__.py:311: in build
    updated_docnames = set(self.read())
../../venvs/everett/lib/python3.7/site-packages/sphinx/builders/__init__.py:418: in read
    self._read_serial(docnames)
../../venvs/everett/lib/python3.7/site-packages/sphinx/builders/__init__.py:439: in _read_serial
    self.read_doc(docname)
../../venvs/everett/lib/python3.7/site-packages/sphinx/builders/__init__.py:479: in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
../../venvs/everett/lib/python3.7/site-packages/sphinx/io.py:221: in read_doc
    pub.publish()
../../venvs/everett/lib/python3.7/site-packages/docutils/core.py:218: in publish
    self.settings)
../../venvs/everett/lib/python3.7/site-packages/sphinx/io.py:126: in read
    self.parse()
../../venvs/everett/lib/python3.7/site-packages/docutils/readers/__init__.py:77: in parse
    self.parser.parse(self.input, document)
../../venvs/everett/lib/python3.7/site-packages/sphinx/parsers.py:102: in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
../../venvs/everett/lib/python3.7/site-packages/docutils/parsers/rst/states.py:171: in run
    input_source=document['source'])
../../venvs/everett/lib/python3.7/site-packages/docutils/statemachine.py:242: in run
    context, state, transitions)
../../venvs/everett/lib/python3.7/site-packages/docutils/statemachine.py:459: in check_line
    return method(match, context, next_state)
../../venvs/everett/lib/python3.7/site-packages/docutils/parsers/rst/states.py:2342: in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
../../venvs/everett/lib/python3.7/site-packages/docutils/parsers/rst/states.py:2354: in explicit_construct
    return method(self, expmatch)
../../venvs/everett/lib/python3.7/site-packages/docutils/parsers/rst/states.py:2097: in directive
    directive_class, match, type_name, option_presets)
../../venvs/everett/lib/python3.7/site-packages/docutils/parsers/rst/states.py:2146: in run_directive
    result = directive_instance.run()
everett/sphinxext.py:417: in run
    self.generate_docs(self.arguments[0], self.content)
everett/sphinxext.py:384: in generate_docs
    docstringlines = prepare_docstring(docstring, ignore=1)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

s = 'This component is the best.\n\n    The best!\n\n    ', ignore = 1, tabsize = 8

    def prepare_docstring(s: str, ignore: int = None, tabsize: int = 8) -> List[str]:
        """Convert a docstring into lines of parseable reST.  Remove common leading
        indentation, where the indentation of a given number of lines (usually just
        one) is ignored.
    
        Return the docstring as a list of lines usable for inserting into a docutils
        ViewList (used as argument of nested_parse().)  An empty line is added to
        act as a separator between this docstring and following content.
        """
        if ignore is None:
            ignore = 1
        else:
            warnings.warn("The 'ignore' argument to parepare_docstring() is deprecated.",
>                         RemovedInSphinx50Warning, stacklevel=2)
E           sphinx.deprecation.RemovedInSphinx50Warning: The 'ignore' argument to parepare_docstring() is deprecated.

../../venvs/everett/lib/python3.7/site-packages/sphinx/util/docstrings.py:61: RemovedInSphinx50Warning
```

This should get fixed.

--- a/everett/sphinxext.py
+++ b/everett/sphinxext.py
@@ -381,7 +381,7 @@ class AutoComponentDirective(Directive):
             docstring_attr = self.options["show-docstring"] or "__doc__"
             docstring = getattr(obj, docstring_attr, None)
             if docstring:
-                docstringlines = prepare_docstring(docstring, ignore=1)
+                docstringlines = prepare_docstring(docstring)
                 for i, line in enumerate(docstringlines):
                     self.add_line(indent + line, sourcename, i)
                 self.add_line("", "")

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -4,8 +4,6 @@ black==20.8b1
 check-manifest==0.44
 flake8==3.8.4
 pytest==6.1.1
-# FIXME(willkg): later versions of Sphinx break the (ridiculous) scaffolding
-# everett does to test the Sphinx extension.
-Sphinx==3.0.4
+Sphinx==3.2.1
 tox==3.20.1
-twine==3.2.0 ; python_version >= '3.6'
+twine==3.2.0

